File: amplify/data/models/UserProfile.ts
import { a } from "@aws-amplify/backend";

export const UserProfileModel = a.model({
  owner: a.string().required(),
  email: a.string().required(),
  firstName: a.string().required(),
  lastName: a.string().required(),
  authenticated: a.boolean(),
  premium: a.boolean(),
  createdAt: a.datetime().required().default(new Date().toISOString()),
  updatedAt: a.datetime().required().default(new Date().toISOString()),
  userBudgets: a.hasMany('UserBudgetItem', 'owner')
})
.authorization(allow => [
  allow.owner().to([
    'read',
    'delete'
  ]),
  allow.group('ADMINS')
])
.identifier(['owner'])

File: AmplifyModels/UserProfile.swift
// swiftlint:disable all
import Amplify
import Foundation

public struct UserProfile: Model {
  public let owner: String
  public var email: String
  public var firstName: String
  public var lastName: String
  public var authenticated: Bool?
  public var premium: Bool?
  public var createdAt: Temporal.DateTime
  public var updatedAt: Temporal.DateTime
  public var userBudgets: List<UserBudgetItem>?
  
  public init(owner: String,
      email: String,
      firstName: String,
      lastName: String,
      authenticated: Bool? = nil,
      premium: Bool? = nil,
      createdAt: Temporal.DateTime,
      updatedAt: Temporal.DateTime,
      userBudgets: List<UserBudgetItem>? = []) {
      self.owner = owner
      self.email = email
      self.firstName = firstName
      self.lastName = lastName
      self.authenticated = authenticated
      self.premium = premium
      self.createdAt = createdAt
      self.updatedAt = updatedAt
      self.userBudgets = userBudgets
  }
}